### DISCUSSION TITLE
Why useFetch hook removal?

### DISCUSSION THREAD

**Discussion by soullivaneuh** [juliencrn/usehooks-ts#567] [Help]
Hi,

I am currently looking at the v3 upgrade guide and saw the `useFetch` hook was considered as deprecated and removed (#503).

You suggest on the guide to rely on dedicated library for that like SWR or react-query.

They are indeed good solutions for complex implementation of data querying like for binding a REST API to a frontend project.

However, `useFetch` can be still useful for light data fetching implementation, for example, when you just need to fetch some public data on the whole project without having to setup a complex library.

The only current "drawback" of your `useFetch` implementation was the JSON response compatibility only (#266), but that may be an easy pick to remove.

This hook is still an enhancement to me as it simplifies the integration of `fetch` inside a react component with `useEffect` and the request state management.

I didn't see any explanation of the reason you decided to remove this hook so this is why I am here, to ask why. 
I'm curious. :wink: 

Cheers and thanks for that library!

**Comment by juliencrn** [comment]
In fact, I haven't considered the "light data fetching implementation" use case. 
This hook was far less effective than solutions like SWR or react-query (user POV) and time-consuming in maintenance (maintainor POV). That's why I removed it.
However, since it has been removed, re-implementing it from scratch will not introduce breaking changes. So I'm open to re-integrate it into the collection. 
It would be nice to not only support GET but all HTTP verbs as an un-opinionated `fetch` wrapper.
